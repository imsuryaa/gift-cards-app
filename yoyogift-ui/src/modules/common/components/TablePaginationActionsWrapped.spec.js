import React from "react";
import { mount } from "enzyme";

import { TablePaginationActions } from "./TablePaginationActionsWrapped";

describe("Footer Test Suite", () => {
  it("should render expected HTML", () => {
    //Mocks
    const classes = { root: { marginTop:"2px" } };
    const theme = {
      direction: "rtl"
    }
    expect(mount(<TablePaginationActions
      classes={classes}
      count={1}
      page={10}
      rowsPerPage={5}
      theme={theme}
    />).html()).toMatchSnapshot();
  });

  it("should render expected HTML", () => {
    //Mocks
    const classes = { root: { marginTop:"2px" } };
    const theme = {
      direction: "sas"
    }
    expect(mount(<TablePaginationActions
      classes={classes}
      count={1}
      page={10}
      rowsPerPage={5}
      theme={theme}
    />).html()).toMatchSnapshot();
  });

  it('should call handleFirstPageButtonClick function', () => {
    //Mocks
    const classes = { root: { marginTop:"2px" } };
    const theme = {
      direction: "sas"
    }
    const wrapper = mount(<TablePaginationActions
      classes={classes}
      count={1}
      page={10}
      rowsPerPage={5}
      theme={theme}
      onChangePage={jest.fn()}
    />);
    const component = wrapper.instance();
    const spy =  jest.spyOn(component, 'handleFirstPageButtonClick');
    
    component.handleFirstPageButtonClick();
   
   expect(component.handleFirstPageButtonClick).toHaveBeenCalled();
   expect(component.handleFirstPageButtonClick).toHaveBeenCalledTimes(1)
  });

  it('should call handleBackButtonClick function', () => {
    //Mocks
    const classes = { root: { marginTop:"2px" } };
    const theme = {
      direction: "sas"
    }
    const wrapper = mount(<TablePaginationActions
      classes={classes}
      count={1}
      page={10}
      rowsPerPage={5}
      theme={theme}
      onChangePage={jest.fn()}
    />);
    const component = wrapper.instance();
    const spy =  jest.spyOn(component, 'handleBackButtonClick');
    
    component.handleBackButtonClick();
   
   expect(component.handleBackButtonClick).toHaveBeenCalled();
   expect(component.handleBackButtonClick).toHaveBeenCalledTimes(1)
  });

  it('should call handleNextButtonClick function', () => {
    //Mocks
    const classes = { root: { marginTop:"2px" } };
    const theme = {
      direction: "sas"
    }
    const wrapper = mount(<TablePaginationActions
      classes={classes}
      count={1}
      page={10}
      rowsPerPage={5}
      theme={theme}
      onChangePage={jest.fn()}
    />);
    const component = wrapper.instance();
    const spy =  jest.spyOn(component, 'handleNextButtonClick');
    
    component.handleNextButtonClick();
   
   expect(component.handleNextButtonClick).toHaveBeenCalled();
   expect(component.handleNextButtonClick).toHaveBeenCalledTimes(1)
  });

  it('should call handleLastPageButtonClick function', () => {
    //Mocks
    const classes = { root: { marginTop:"2px" } };
    const theme = {
      direction: "sas"
    }
    const wrapper = mount(<TablePaginationActions
      classes={classes}
      count={1}
      page={10}
      rowsPerPage={5}
      theme={theme}
      onChangePage={jest.fn()}
    />);
    const component = wrapper.instance();
    const spy =  jest.spyOn(component, 'handleLastPageButtonClick');
    
    component.handleLastPageButtonClick();
   
   expect(component.handleLastPageButtonClick).toHaveBeenCalled();
   expect(component.handleLastPageButtonClick).toHaveBeenCalledTimes(1)
  });
});