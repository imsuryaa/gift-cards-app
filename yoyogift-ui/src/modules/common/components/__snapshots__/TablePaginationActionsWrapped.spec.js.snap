// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Test Suite should render expected HTML 1`] = `"<div class=\\"[object Object]\\"><button class=\\"MuiButtonBase-root-7 MuiIconButton-root-1\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"First Page\\"><span class=\\"MuiIconButton-label-6\\"><svg class=\\"MuiSvgIcon-root-10\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M5.59 7.41L10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0V0z\\"></path></svg></span><span class=\\"MuiTouchRipple-root-19\\"></span></button><button class=\\"MuiButtonBase-root-7 MuiIconButton-root-1\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"Previous Page\\"><span class=\\"MuiIconButton-label-6\\"><svg class=\\"MuiSvgIcon-root-10\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0V0z\\"></path></svg></span><span class=\\"MuiTouchRipple-root-19\\"></span></button><button class=\\"MuiButtonBase-root-7 MuiButtonBase-disabled-8 MuiIconButton-root-1 MuiIconButton-disabled-5\\" tabindex=\\"-1\\" type=\\"button\\" disabled=\\"\\" aria-label=\\"Next Page\\"><span class=\\"MuiIconButton-label-6\\"><svg class=\\"MuiSvgIcon-root-10\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0V0z\\"></path></svg></span></button><button class=\\"MuiButtonBase-root-7 MuiButtonBase-disabled-8 MuiIconButton-root-1 MuiIconButton-disabled-5\\" tabindex=\\"-1\\" type=\\"button\\" disabled=\\"\\" aria-label=\\"Last Page\\"><span class=\\"MuiIconButton-label-6\\"><svg class=\\"MuiSvgIcon-root-10\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M18.41 16.59L13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z\\"></path><path fill=\\"none\\" d=\\"M24 24H0V0h24v24z\\"></path></svg></span></button></div>"`;

exports[`Footer Test Suite should render expected HTML 2`] = `"<div class=\\"[object Object]\\"><button class=\\"MuiButtonBase-root-7 MuiIconButton-root-1\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"First Page\\"><span class=\\"MuiIconButton-label-6\\"><svg class=\\"MuiSvgIcon-root-10\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M18.41 16.59L13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z\\"></path><path fill=\\"none\\" d=\\"M24 24H0V0h24v24z\\"></path></svg></span><span class=\\"MuiTouchRipple-root-19\\"></span></button><button class=\\"MuiButtonBase-root-7 MuiIconButton-root-1\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"Previous Page\\"><span class=\\"MuiIconButton-label-6\\"><svg class=\\"MuiSvgIcon-root-10\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0V0z\\"></path></svg></span><span class=\\"MuiTouchRipple-root-19\\"></span></button><button class=\\"MuiButtonBase-root-7 MuiButtonBase-disabled-8 MuiIconButton-root-1 MuiIconButton-disabled-5\\" tabindex=\\"-1\\" type=\\"button\\" disabled=\\"\\" aria-label=\\"Next Page\\"><span class=\\"MuiIconButton-label-6\\"><svg class=\\"MuiSvgIcon-root-10\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0V0z\\"></path></svg></span></button><button class=\\"MuiButtonBase-root-7 MuiButtonBase-disabled-8 MuiIconButton-root-1 MuiIconButton-disabled-5\\" tabindex=\\"-1\\" type=\\"button\\" disabled=\\"\\" aria-label=\\"Last Page\\"><span class=\\"MuiIconButton-label-6\\"><svg class=\\"MuiSvgIcon-root-10\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M5.59 7.41L10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0V0z\\"></path></svg></span></button></div>"`;
